R/retreive-osm.R

```diff
@@ -43,11 +29,54 @@ osmdata_as_sf <- function(key, value, bb) {
 #' @examples
 #' get_osmdata("Bucharest", "waterway", "river")
 get_osmdata <- function(name, key, value) {
-  bb <- CRiSp::osm_bb(name)
+  bb <- osmdata::getbb(name)
   CRiSp::osmdata_as_sf(key, value, bb)
 }
 
 name <- NULL
+#' Get the city boundary from OpenStreetMap
+#'
+#' @param city_name A character string with the name of the city
+#'
+#' @return An sf object with the city boundary
+#' @export
+#'
+#' @examples
+#' get_osm_city_boundary("Bucharest")
+get_osm_city_boundary <- function(city_name) {
+  city_boundary <- tryCatch(
+    {
+      CRiSp::get_osmdata(city_name, "place", "city")$osm_multipolygons$geometry
```

Comment:
After trying to run this for Berlin:
```
Berlin <- CRiSp::get_osm_city_boundary("Berlin")
```
I got two features as output, with the two geometries representing administrative boundaries for Berlin and Postdam:
```
# Berlin
Geometry set for 2 features 
Geometry type: MULTIPOLYGON
Dimension:     XY
Bounding box:  xmin: 12.88678 ymin: 52.33824 xmax: 13.76116 ymax: 52.67551
Geodetic CRS:  WGS 84
MULTIPOLYGON (((13.09076 52.41156, 13.09144 52....
MULTIPOLYGON (((13.15028 52.56771, 13.15014 52....
```
![plot](https://github.com/user-attachments/assets/8da05e5e-ece4-453e-9b0f-a2dd5fdf6aab)

Filtering the dataframe here seems solve the issue for this specific case, but I am not sure how robust this is (and we are somehow enforcing the use of English names for the cities)?

```suggestion
      CRiSp::get_osmdata(city_name, "place", "city")$osm_multipolygons |> dplyr::filter(.data$`name:en` == city_name) |> st_geometry()
```

Reply:
This is an important edge case, I am sure we will come across this again. Further filtering can be made inside the function. I'll try to update the function to return the correct boundary for Berlin as well.

Reply:
I updated the function to work for the Berlin case. I made the function more compact and a bit more robust; it can be further improved, but that can be done in a separate PR. I will merge this one.